include/tvm/runtime/ndarray.h

```diff
@@ -281,6 +284,10 @@ class NDArray::Container {
   int32_t array_type_code_{0};
   /*! \brief The internal reference counter */
   std::atomic<int> ref_counter_{0};
+
+  /*! \brief Buffer allocated by allocator */
+  Buffer* buffer_;
```

Comment:
hide the buffer pointer in the ctx field instead of introducing a new field

Reply:
@icemelon9 could you follow up on this comment since the change is from you?